### ISSUE TITLE
The USB set address request failed.

### ISSUE THREAD

**Issue by zgdump** [mvirkkunen/usbd-serial#24]
Hi!

I used your library in my project for STM32F411CEU and noticed that sometimes Windows shows error 43 in Device Manager. If reconnect device, then everything works as it should

To reproduce the issue, I plugged and popped the device several times. Sometimes it also occurred if leave the stm32 connected via USB and flash using st-link

![image](https://user-images.githubusercontent.com/60278551/147861074-55114e65-fe88-40b9-af4d-00be46e230f1.png)

**Comment by zgdump** [comment]
My code is based on example...

```rust
static mut EP_MEMORY: [u32; 1024] = [0; 1024];

#[entry]
fn main() -> ! {
    rtt_init_print!();
    rprintln!("Hello!");

    let dp = pac::Peripherals::take().unwrap();

    let rcc = dp.RCC.constrain();

    let clocks = rcc
        .cfgr
        .use_hse(25.mhz())
        .sysclk(48.mhz())
        .require_pll48clk()
        .freeze();

    let gpioa = dp.GPIOA.split();
    let usb = USB {
        usb_global: dp.OTG_FS_GLOBAL,
        usb_device: dp.OTG_FS_DEVICE,
        usb_pwrclk: dp.OTG_FS_PWRCLK,
        pin_dm: gpioa.pa11.into_alternate(),
        pin_dp: gpioa.pa12.into_alternate(),
        hclk: clocks.hclk(),
    };

    let mut usb_bus = UsbBus::new(usb, unsafe { &mut EP_MEMORY });

    let mut serial = SerialPort::new(&usb_bus);

    let mut usb_dev = UsbDeviceBuilder::new(&usb_bus, UsbVidPid(0x16c0, 0x27dd))
        .product("Serial port")
        .device_class(USB_CLASS_CDC)
        .build();

    loop {
        if !usb_dev.poll(&mut [&mut serial]) {
            continue;
        }

        let mut buf = [0u8; 64];

        match serial.read(&mut buf[..]) {
            Ok(count) => {}
            Err(UsbError::WouldBlock) => {}
            Err(err) => {}
        };

        match serial.write(&[0x3a, 0x29]) {
            Ok(count) => {}
            Err(UsbError::WouldBlock) => {}
            Err(err) => {}
        };
    }
}
```

**Comment by zgdump** [comment]
Experimenting, I noticed that with an increase in frequency to 100mhz, the problem is no longer reproduced. And if you lower it to 25, then there is almost always an error when connecting.

**Comment by zgdump** [comment]
I get dump from the usb using Wireshark. This dump contains errors and successful connection. STM32 freq is 25mhz

[Uploading wireshark-dump.zip…]()

**Comment by mvirkkunen** [comment]
By increasing the frequency, did you mean the sysclk frequency? Which hardware are you using? Have you tried running it in release mode just in case (although in this case it really shouldn't matter)?

The dump didn't seem to contain anything enumeration related, just a bunch of interrupt requests from another USB devide. At any rate this is very unlikely to be related to `usbd-serial` if the device doesn't even manage to enumerate, it's either got something to do with `usb-device` or `synopsys-usb-otg`.

**Comment by zgdump** [comment]
@mvirkkunen 

Yes, I mean sysclk. My chip is STM32F411CEU6. And yes, I always run in release

Sorry, initially I took the dump incorrectly using wireshark. Recently updated the archive, there is a more correct dump. Although, nothing concrete and useful... I will try to write to the usb repository. If you don't have the error, then I'm sorry for the trouble

**Comment by mvirkkunen** [comment]
I was wondering about the entire hardware, is it custom or pre-existing board? Could it possibly have a power-on delay that causes an intermittent fault?

I can see the update dump now and it doesn't contain errors, it's possible that your recording setup doesn't see that type of low-level error.

`usb-device` is my repository as well so I'll just move this issue.
